apps/rush/src/taskRunner/TaskRunner.ts

```diff
@@ -280,10 +282,27 @@ export default class TaskRunner {
   }
 
   /**
-   * Calculate the number of packages which must be build before we reach
+   * Checks for projects that indirectly depend on themselves.
+   */
+  private _checkForCyclicDependencies(tasks: ReadonlyArray<ITask>, dependencyChain: string[]): void {
+    for (const task of tasks) {
+      if (dependencyChain.indexOf(task.name) >= 0) {
+        throw new Error('A cyclic dependency was encountered:' + os.EOL
+          + '  ' + [...dependencyChain, task.name].reverse().join('\n  -> ') + os.EOL
```

Comment:
>\n [](start = 67, length = 2)

Sorta odd that you're using `os.EOL` everywhere else.

Reply:
Hmmm...  we never established any proper guidance for os.EOL.

My personal opinions:

- Identical behavior on across operating systems is extremely important

- When we save a file to disk, it should be "\r\n", since nearly every viewer/editor can understand that

- When we output to the console, os.EOL would be better

- In keeping with the general principle of encodings, EOL conversions should be applied only at last step before a value is written.  Internally our code should use "\n" everywhere because a single character is easier to work with, and likely to be used by OSS libraries.  

---
In reply to: [143555699](https://github.com/Microsoft/web-build-tools/pull/390#discussion_r143555699) [](ancestors = 143555699)

Reply:
That said, I made them consistent

---
In reply to: [143559149](https://github.com/Microsoft/web-build-tools/pull/390#discussion_r143559149) [](ancestors = 143559149,143555699)

Reply:
Cool.

---
In reply to: [143559149](https://github.com/Microsoft/web-build-tools/pull/390#discussion_r143559149) [](ancestors = 143559149,143555699)